src/main/scala/org/broadinstitute/dsde/workbench/sam/config/TermsOfServiceConfig.scala

```diff
@@ -1,12 +1,27 @@
 package org.broadinstitute.dsde.workbench.sam.config
 
+import java.time.Instant
+
 /** Terms of Service configuration.
   * @param isGracePeriodEnabled
   *   Set to true if the grace period for ToS acceptance is active
   * @param version
   *   The latest version of the Terms of Service
-  * @param url
+  * @param baseUrl
   *   The url to the Terra Terms of Service. Used for validation and will be displayed to user in error messages
+  * @param rollingAcceptanceWindowExpiration
+  *   The expiration time for the rolling acceptance window. If the user has not accepted the new ToS by this time, they will be denied access to the system.
+  *   Must be a valid UTC datetime string in ISO 8601 format example: 2007-12-03T10:15:30.00Z
+  *
+  * @param rollingAcceptanceWindowPreviousTosVersion
+  *   The previous version of the ToS that the user must have accepted in order to be granted access to the system under the rolling window.
   */
 
-case class TermsOfServiceConfig(isTosEnabled: Boolean, isGracePeriodEnabled: Boolean, version: String, baseUrl: String)
+case class TermsOfServiceConfig(
+    isTosEnabled: Boolean,
+    isGracePeriodEnabled: Boolean,
+    version: String,
+    baseUrl: String,
+    rollingAcceptanceWindowExpiration: Instant,
```

Comment:
We might want to make this `Option[Instant]`. Seems like we could skip a bunch of logic if we want to release Sam without a new ToS by setting it to `None`.